src/core/components/dag.js

```diff
@@ -9,6 +9,21 @@ const flattenDeep = require('lodash.flattendeep')
 module.exports = function dag (self) {
   return {
     put: promisify((dagNode, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+      } else if (options.cid && options.format && options.hashAlg) {
+        throw new Error('Can\'t put dag node. Please provide either `cid` OR `format` and `hashAlg` options.');
+      } else if ((options.format && !options.hashAlg) || (!options.format && options.hashAlg)) {
+        throw new Error('Can\'t put dag node. Please provide `format` AND `hashAlg` options.');
```

Comment:
For a moment I was confused whether throwing synchronously inside `promisify()` makes sense, but I don't seem to have a handle to "reject" it either.

Looking at its testsuite, it seems to be the way to go: https://github.com/manuel-di-iorio/promisify-es6/blob/master/test.js#L79

Reply:
I would replace `throw new ...` with `return callback(new ...)`